src/generateTypeFiles.js

```diff
@@ -32,41 +35,33 @@ export const generateTypeFiles = (
     document: DocumentNode,
     options: Options,
 ) => {
-    const {documentToFlowTypes} = require('.');
-    const path = require('path');
-    const fs = require('fs');
+    const generatedDir = path.join(path.dirname(fileName), '__generated__');
+    const indexFile = path.join(generatedDir, 'index.js');
 
-    const indexFile = (generatedDir) => path.join(generatedDir, 'index.js');
+    if (!fs.existsSync(generatedDir)) {
+        fs.mkdirSync(generatedDir, {recursive: true});
 
-    const maybeCreateGeneratedDir = (generatedDir) => {
-        if (!fs.existsSync(generatedDir)) {
-            fs.mkdirSync(generatedDir, {recursive: true});
+        // Now write an index.js for each __generated__ dir.
+        fs.writeFileSync(indexFile, indexPrelude(options.regenerateCommand));
+    }
 
-            // Now write an index.js for each __generated__ dir.
-            fs.writeFileSync(
-                indexFile(generatedDir),
-                indexPrelude(options.regenerateCommand),
-            );
-        }
-    };
+    let indexContents = fs.readFileSync(indexFile, 'utf8');
 
     /// Write export for __generated__/index.js if it doesn't exist
-    const writeToIndex = (filePath, typeName) => {
-        const index = indexFile(path.dirname(filePath));
-        const indexContents = fs.readFileSync(index, 'utf8');
+    const addToIndex = (filePath, typeName) => {
         const newLine = `export type {${typeName}} from './${path.basename(
             filePath,
         )}';`;
-        if (indexContents.indexOf(path.basename(filePath)) === -1) {
-            fs.appendFileSync(index, newLine + '\n');
+        if (indexContents.indexOf('./' + path.basename(filePath)) === -1) {
+            indexContents += newLine + '\n';
         } else {
             const lines = indexContents.split('\n').map((line) => {
-                if (line.includes(path.basename(filePath))) {
+                if (line.includes('./' + path.basename(filePath))) {
                     return newLine;
                 }
                 return line;
             });
-            fs.writeFileSync(index, lines.join('\n'));
+            indexContents = lines.join('\n');
```

Comment:
Was just thinking about the hoops we're jumping through here.  I wonder if it'd be simpler to just have a `Set` and then just map it out to the file at the end. 🤔 

No action required.

Reply:
true story